mono/metadata/domain-internals.h

```diff
@@ -235,11 +235,16 @@ struct _MonoJitInfo {
 
 #define MONO_SIZEOF_JIT_INFO (offsetof (struct _MonoJitInfo, clauses))
 
+typedef struct {
+	gpointer *static_data;
+	uint32_t gc_handle;
+} ContextStaticData;
+
 struct _MonoAppContext {
 	MonoObject obj;
 	gint32 domain_id;
 	gint32 context_id;
-	gpointer *static_data;
+	ContextStaticData *data;
```

Comment:
Oh! Now I see the issue, didn't realize static_data was mutated after the initial setup.

Anyways, the current approach of mono_gc_alloc_fixed with MONO_GC_DESCRIPTOR_NULL is bad and should be avoided as it forces sgen to conservatively scan ContextStaticData (which is expensive).

What about extending context to hold pointers to static_data and ContextStaticData? This way we can update the later when expanding static_data.